components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1186,23 +1187,34 @@ where
                 self.on_raft_log_gc_flushed();
             }
             SignificantMsg::RaftlogFetched { context, res } => {
-                let low = res.low;
-                if self.fsm.peer.term() != res.term {
-                    self.fsm.peer.mut_store().clean_async_fetch_res(low);
-                } else {
-                    self.fsm
-                        .peer
-                        .mut_store()
-                        .update_async_fetch_res(low, Some(res));
-                }
-                self.fsm.peer.raft_group.on_entries_fetched(context);
-                // clean the async fetch result immediately if not used to free memory
-                self.fsm.peer.mut_store().update_async_fetch_res(low, None);
-                self.fsm.has_ready = true;
+                self.on_raft_log_fetched(context, res);
             }
         }
     }
 
+    fn on_raft_log_fetched(&mut self, context: GetEntriesContext, res: Box<RaftlogFetchResult>) {
+        let low = res.low;
+        // if the peer is being destroyed, ignore the result.
+        if self.fsm.peer.pending_remove {
```

Comment:
Seems it's a subset of https://github.com/tikv/tikv/issues/12026. Can you also fix #12026 in this PR?